File: asc/semis_reports/report/sample_test_report/sample_test_report.js
// Copyright (c) 2016, Frappe Technologies and contributors
// For license information, please see license.txt
/* eslint-disable */

frappe.query_reports["Sample test report"] = {
	"filters": [{
		fieldname: "number_filter",
		label: "Number Filter",
		fieldtype: "Select",
		//You can supply the options as a string of new-line (\n) separated values,
		//    or as an array of strings such as options: ["1","2","3","4","5","6","7"],
		options: "1\n2\n3\n4\n5\n6\n7",
		default: 3

	},
	{
		fieldname: "date_filter",
		label: "Date Filter",
		fieldtype: "Date",
		//Note the following default attribute, which contains an API call
		default: frappe.datetime.get_today()
	},
	{
		fieldname: "check_filter",
		label: "Check Filter",
		fieldtype: "Check",
		default: 1,
	},
	{
		fieldname: "user_filter",
		label: "User Filter",
		fieldtype: "Link",
		options: "User",
		reqd: 0,
	}
	]
};

File: asc/semis_reports/report/sample_test_report/sample_test_report.py
# Copyright (c) 2013, Frappe Technologies and contributors
# For license information, please see license.txt

# import frappe
import frappe
import datetime
def execute(filters=None):
    #frappe.msgprint("<pre>{}</pre>".format(filters))
    columns = [
        {'fieldname':'name','label':'ID','fieldtype':'Data'},
        {'fieldname':'first_name','label':'First Name','fieldtype':'Data'},
        {'fieldname':'last_name','label':'Last Name','fieldtype':'Data'},
        {'fieldname':'creation','label':'Creation','fieldtype':'Date'}
    ]
    data = frappe.db.get_all('User', ['name','first_name','last_name','creation'])
    #frappe.msgprint("<span style='color:Red;'>Once this popup has served it's purpose, then comment out it's invocation, viz. #frappe.msgprint...</span><br><br>" + "<pre>{}</pre>".format(frappe.as_json(data)))
    datefilter = datetime.datetime.strptime(filters.date_filter,"%Y-%m-%d").date()
    today = datetime.datetime.now(tz=None).date()
    data = [dic for dic in data if dic.creation.date() > datefilter]
    data = sorted(data, key=lambda k: k['first_name'])
    chart = {
        'title':"Script Chart Tutorial : Days since the user's database record was created",
        'data':{
            'labels':[str(dic.first_name) + " " + str(dic.last_name) for dic in data],
            'datasets':[{'values':[(today - dic.creation.date()).days for dic in data]}]
        },
        'type':'line',
        'height':300,
        'colors':['#F16A61'],
        'lineOptions':{'hideDots':0, 'dotSize':6, 'regionFill':1}
    }
    report_summary = [{"label":"Count","value":len(data),'indicator':'Red' if len(data) < 10 else 'Green'}]
    return columns, data, None, chart, report_summary
